vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -279,57 +318,96 @@ public void add(List<Document> documents) {
+				.bodyValue(embeddingsJson)
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("Vector index does not exist");
+		}
 	}
 
 	@Override
 	public Optional<Boolean> delete(List<String> idList) {
-		throw new UnsupportedOperationException("This vector store doesn't support delete (yet)");
+		try {
+			client.method(HttpMethod.DELETE)
+				.uri("/" + INDEX_NAME + "/embeddings/")
+				.body(BodyInserters.fromValue(idList))
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("Error removing embedding with ID: " + e);
+		}
+		return Optional.of(true);
 	}
 
 	@Override
 	public List<Document> similaritySearch(SearchRequest request) {
-		List<Double> vector = this.embeddingClient.embed(request.getQuery());
-
-		return client.post()
-			.uri("/query")
-			.contentType(MediaType.APPLICATION_JSON)
-			.bodyValue(new QueryRequest(vector, request.getTopK(), topKPerBucket, true))
-			.retrieve()
-			.bodyToFlux(QueryResponse.class)
-			.filter(r -> r.score >= request.getSimilarityThreshold())
-			.map(r -> {
-				Map<String, Object> metadata = r.metadata;
-				metadata.put(DISTANCE_METADATA_FIELD_NAME, 1 - r.score);
-				String content = (String) metadata.remove(documentField);
-
-				return new Document(r.key, content, metadata);
-			})
-			.collectList()
-			.block();
+		try {
+			List<Double> vector = this.embeddingClient.embed(request.getQuery());
+			List<Float> floatVector = vector.stream().map(Double::floatValue).toList();
+
+			return client.post()
+				.uri("/" + INDEX_NAME + "/query")
+				.contentType(MediaType.APPLICATION_JSON)
+				.bodyValue(new QueryRequest(floatVector, request.getTopK(), topKPerBucket, true))
+				.retrieve()
+				.bodyToFlux(QueryResponse.class)
+				.filter(r -> r.score >= request.getSimilarityThreshold())
+				.map(r -> {
+					Map<String, Object> metadata = r.metadata;
+					metadata.put(DISTANCE_METADATA_FIELD_NAME, 1 - r.score);
+					String content = (String) metadata.remove(documentField);
+
+					return new Document(r.key, content, metadata);
+				})
+				.collectList()
+				.block();
+		}
+		catch (Exception e) {
+			throw new RuntimeException("Error in similarity search", e);
+		}
 	}
 
-	public void createIndex() {
-		client.post()
-			.contentType(MediaType.APPLICATION_JSON)
-			.bodyValue(new CreateRequest())
-			.retrieve()
-			.bodyToMono(Void.class)
-			.block();
+	public void createIndex() throws JsonProcessingException {
+		try {
+			CreateRequest createRequest = new CreateRequest(INDEX_NAME);
+			ObjectMapper objectMapper = new ObjectMapper();
+			String index = objectMapper.writeValueAsString(createRequest);
+			client.post()
+				.contentType(MediaType.APPLICATION_JSON)
+				.bodyValue(index)
+				.retrieve()
+				.bodyToMono(Void.class)
+				.block();
+		}
+		catch (WebClientResponseException exception) {
+			logger.warn("Index already exists", exception);
```

Comment:
create index is not failing here. it will just log a warning message and move on. I have not added any security exceptions. Don't know if we need to bring security into the scope of this PR.

Reply:
When does a webclient response exception get thrown?  What i meant with the question was, does a security exception also manifest as one of these?
Are there other cases where it's not due to an index already being present?

Reply:
WebclientResponseException will throw an exception at any http non-success code. Security exception will also manifest as one of those like 401 or 403. What we can do here is that we can generalize this exception.